test/framework/toy_build.py

```diff
@@ -3740,6 +3745,37 @@ def test_toy_post_install_patches(self):
         # verify whether patch indeed got applied
         self.assertEqual(toy_readme_txt, "toy 0.0, a toy test program\n")
 
+    def test_toy_unavailable_os_dep(self):
+        """
+        Test unavailable OS dep
+        """
+        test_ecs = os.path.join(os.path.dirname(__file__), 'easyconfigs', 'test_ecs')
+        toy_ec = os.path.join(test_ecs, 't', 'toy', 'toy-0.0.eb')
+
+        test_ec_txt = read_file(toy_ec)
+        test_ec_txt += "\nosdependencies = [('package-does-not-exist')]"
```

Comment:
Hmm, seems a bit strange to trigger a "parse" error by injecting a non-existing OS dependency.

How about just injecting some junk that will definitely trigger a *parse* error?

Something like:

```python
# add line with incorrect syntax (closing '{' with ']')
test_ec_txt += "\nfoo = {'bar': 1]"
```

and then look for "`Parsing easyconfig file failed`" error

Reply:
I explicitly wanted to catch the case of a missing OS and check that we generate a test report. An incorrectly formatted easyconfig will be picked up elsewhere, such as by the easyconfigs test suite. However, missing an OS dependency still leaves a valid ec, with the missing OS dependency gets detected when it is parsed.

Reply:
OK, makes sense.

Maybe it's worth adding a comment on that in this test?

Reply:
4d887a9